README.md

```diff
@@ -25,10 +25,31 @@ fn main() {
 }
 ```
 
+Or using question mark operator:
+
+```rust
+use std::io::{stdout, Write};
+use curl::easy::{Easy, WriteError};
+use std::error::Error;
+
+fn main() -> Result<(), Box<dyn Error>> {
+    let mut easy = Easy::new();
+    easy.url("https://www.rust-lang.org/")?;
+    easy.write_function(|data| {
+        stdout().write_all(data).or(Err(WriteError::Pause))
```

Comment:
This doesn't compile because `write_all` returns `()` on success, not `usize`. Try this instead:

```suggestion
        stdout().write_all(data)
            .map(|_| data.len())
            .map_err(|_| WriteError::Pause)
```